### ISSUE TITLE
Error for Composer on a New Drupal 9 Pantheon Site

### ISSUE THREAD

**Issue by leorawe** [lando/pantheon#95]
I have a new Pantheon Drupal 9 site. Here is the .lando.yml:
```
name: (...)
recipe: pantheon
config:
  framework: drupal8
  webroot: web
  site: (....)
  id: (...)
```
When I build or rebuild, I get this error (debug output):
`ERROR ==> Fatal error: require(): Failed opening required '/app/vendor/composer/../../web/core/includes/bootstrap.inc' (include_path='/app/vendor/pear/archive_tar:/app/vendor/pear/console_getopt:/app/vendor/pear/pear-core-minimal/src:/app/vendor/pear/pear_exception:.:/usr/local/lib/php') in /app/vendor/composer/autoload_real.php on line 75
`
At Dustin LeBlanc's suggestion, I took **drush_version: 10** out of pantheon.upstream.yml -- but I still got the same error.

The build gives these warnings:
```
All settings correct for using Composer
Downloading...
Composer (version 2.0.7) successfully installed to: /usr/local/bin/composer
Use it: php /usr/local/bin/composer
PHP Warning:  require(/app/vendor/composer/../../web/core/includes/bootstrap.inc): failed to open stream: No such file or directory in /app/vendor/composer/autoload_real.php on line 75
```
Then it stops building at usermod: no changes.

More of the DEBUG:
```

DEBUG ==>  message=Fatal error: require(): Failed opening required '/app/vendor/composer/../../web/core/includes/bootstrap.inc' (include_path='/app/vendor/pear/archive_tar:/app/vendor/pear/console_getopt:/app/vendor/pear/pear-core-minimal/src:/app/vendor/pear/pear_exception:.:/usr/local/lib/php') in /app/vendor/composer/autoload_real.php on line 75, stack=Error: Fatal error: require(): Failed opening required '/app/vendor/composer/../../web/core/includes/bootstrap.inc' (include_path='/app/vendor/pear/archive_tar:/app/vendor/pear/console_getopt:/app/vendor/pear/pear-core-minimal/src:/app/vendor/pear/pear_exception:.:/usr/local/lib/php') in /app/vendor/composer/autoload_real.php on line 75
    at /snapshot/cli/lib/shell.js
From previous event:
    at Shell.sh (/snapshot/cli/lib/shell.js)
    at Object.dc (/snapshot/cli/lib/bootstrap.js)
    at compose (/snapshot/cli/lib/bootstrap.js)
From previous event:
    at /snapshot/cli/lib/router.js
    at processImmediate (internal/timers.js:464:21)
From previous event:
    at Object.run (/snapshot/cli/lib/router.js)
    at run (/snapshot/cli/lib/engine.js)
    at /snapshot/cli/lib/router.js
From previous event:
    at Object.eventWrapper (/snapshot/cli/lib/router.js)
    at Engine.engineCmd (/snapshot/cli/lib/engine.js)
    at Engine.run (/snapshot/cli/lib/engine.js)
    at Object.runBuild (/snapshot/cli/plugins/lando-services/lib/utils.js)
    at AsyncEvents.<anonymous> (/snapshot/cli/plugins/lando-services/app.js)
    at AsyncEvents.handle (/snapshot/cli/lib/events.js)
    at /snapshot/cli/lib/events.js
From previous event:
    at AsyncEvents.emit (/snapshot/cli/lib/events.js)
    at /snapshot/cli/lib/app.js
From previous event:
    at App.start (/snapshot/cli/lib/app.js)
    at Object.run (/snapshot/cli/plugins/lando-core/tasks/start.js)
    at /snapshot/cli/lib/cli.js
    at processImmediate (internal/timers.js:464:21)
From previous event:
    at /snapshot/cli/lib/cli.js
From previous event:
    at Object.handler (/snapshot/cli/lib/cli.js)
    at Object.runCommand (/snapshot/cli/node_modules/yargs/lib/command.js:238:44)
    at Object.parseArgs [as _parseArgs] (/snapshot/cli/node_modules/yargs/yargs.js:1063:30)
    at Function.get [as argv] (/snapshot/cli/node_modules/yargs/yargs.js:1004:21)
    at Cli.init (/snapshot/cli/lib/cli.js)
    at Cli.run (/snapshot/cli/lib/cli.js)
From previous event:
    at /snapshot/cli/bin/lando.js
From previous event:
    at Object.<anonymous> (/snapshot/cli/bin/lando.js)
    at Module._compile (pkg/prelude/bootstrap.js:1751:22)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Function.runMain (pkg/prelude/bootstrap.js:1804:12)
    at internal/main/run_main_module.js:17:47, __stackCleaned__=true
ruspace 15:53:12




```

**Comment by labboy0276** [comment]
@leorawe what version of Lando are you on currently?

**Comment by leorawe** [comment]
Lando v3.6.0

**Comment by labboy0276** [comment]
@leorawe ok, did you try purging your `~/.lando` folder and doing a `lando destroy` on the app you are having an issue with?

**Comment by leorawe** [comment]
I did not purge my ~/.lando -- I also just noticed it had the same name as my older lando image, which may have caused conflicts. I am going to try this out sometime to day, and I will report back here.

**Comment by labboy0276** [comment]
@leorawe is that the full .lando.yml you have?  

Also what does your composer.json look like?  I am curious if this is an app issue itself.

**Comment by leorawe** [comment]
Here is the composer.json:
```
# #
{
    "name": "pantheon-upstreams/drupal-project",
    "description": "Install Drupal 9 with Composer on Pantheon.",
    "type": "project",
    "license": "MIT",
    "repositories": [
        {
            "type": "composer",
            "url": "https://packages.drupal.org/8"
        },
        {
            "type": "path",
            "url": "upstream-configuration"
        }
    ],
    "require": {
        "pantheon-upstreams/upstream-configuration": "self.version",
        "composer/installers": "^1.9",
        "drupal/core-composer-scaffold": "^9.2",
        "drupal/core-recommended": "^9.2",
        "pantheon-systems/drupal-integrations": "^9",
        "cweagans/composer-patches": "^1.7",
        "drush/drush": "^10"
    },
    "require-dev": {
        "drupal/core-dev": "^9.2"
    },
    "conflict": {
            "drupal/drupal": "*"
    },
    "minimum-stability": "stable",
    "prefer-stable": true,
    "extra": {
        "drupal-scaffold": {
            "locations": {
                "web-root": "./web"
            },
            "allowed-packages": [
                "pantheon-systems/drupal-integrations"
            ],
            "file-mapping": {
                "[project-root]/.editorconfig": false,
                "[project-root]/pantheon.upstream.yml": false,
                "[project-root]/.gitattributes": false
            }
        },
        "installer-paths": {
            "web/core": ["type:drupal-core"],
            "web/libraries/{$name}": ["type:drupal-library"],
            "web/modules/contrib/{$name}": ["type:drupal-module"],
            "web/profiles/contrib/{$name}": ["type:drupal-profile"],
            "web/themes/contrib/{$name}": ["type:drupal-theme"],
            "drush/Commands/contrib/{$name}": ["type:drupal-drush"],
            "web/modules/custom/{$name}": ["type:drupal-custom-module"],
            "web/profiles/custom/{$name}": ["type:drupal-custom-profile"],
            "web/themes/custom/{$name}": ["type:drupal-custom-theme"],
            "web/private/scripts/quicksilver/{$name}/": ["type:quicksilver-script"]
        },
        "composer-exit-on-patch-failure": true,
        "patchLevel": {
            "drupal/core": "-p2"
        }
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "platform": {
            "php": "7.4"
        }
    }
}

```

This is a brand new Drupal 9 Pantheon site. I have not yet added in the modules and theme.

**Comment by leorawe** [comment]
And yes, there is nothing else yet to the .lando.yml file. I would love help added more information to it, if needed.

**Comment by labboy0276** [comment]
@leorawe ok thank you.  Out of curiosity are the `web/core/includes/bootstrap.inc` file there in the code base?

Could you also run a  `lando composer install` just to be safe.

**Comment by labboy0276** [comment]
if you are allowed, could you add me to the pantheon site and I can take a look and see @leorawe ?  My pantheon email is john@thinktandem.io

**Comment by leorawe** [comment]
You are now added.

**Comment by labboy0276** [comment]
Thansk @leorawe it seems ok on my end so far:

```
john@tandem:~/lando/1-test-sites/rutgers-njspace (master)$ lando rebuild
? Are you sure you want to rebuild? Yes
Rising anew like a fire phoenix from the ashes! Rebuilding app...
No stopped containers
Pulling database  ... done
Pulling appserver ... done
Pulling cache     ... done
Pulling edge      ... done
Pulling index     ... done
database uses an image, skipping
index uses an image, skipping
appserver uses an image, skipping
cache uses an image, skipping
appserver_nginx uses an image, skipping
edge uses an image, skipping
edge_ssl uses an image, skipping
Recreating landoproxyhyperion5000gandalfedition_proxy_1 ... done
Creating network "REDACTED_default" with the default driver
Creating volume "REDACTED_data_appserver_nginx" with default driver
Creating volume "REDACTED_home_appserver_nginx" with default driver
Creating volume "REDACTED_data_appserver" with default driver
Creating volume "REDACTED_home_appserver" with default driver
Creating volume "REDACTED_data_database" with default driver
Creating volume "REDACTED_home_database" with default driver
Creating volume "REDACTED_data_cache" with default driver
Creating volume "REDACTED_home_cache" with default driver
Creating volume "REDACTED_data_edge" with default driver
Creating volume "REDACTED_home_edge_ssl" with default driver
Creating volume "REDACTED_home_edge" with default driver
Creating volume "REDACTED_data_index" with default driver
Creating volume "REDACTED_home_index" with default driver
Creating REDACTED_appserver_1 ... done
usermod: no changes
usermod: no changes
Pantheon pre-run scripting
Generating RSA private key, 2048 bit long modulus (2 primes)
.........................................+++++
............................................................+++++
e is 65537 (0x010001)
Signature ok
subject=C = US, ST = California, L = San Francisco, O = Lando, OU = Bespin, CN = REDACTED
Getting CA Private Key
All settings correct for using Composer
Downloading...

Composer (version 2.0.7) successfully installed to: /usr/local/bin/composer
Use it: php /usr/local/bin/composer

Changed current directory to /var/www/.composer
The "hirak/prestissimo" plugin was skipped because it requires a Plugin API version ("^1.0.0") that does not match your Composer installation ("2.0.0"). You may need to run composer update with the "--no-plugins" option.
./composer.json has been updated
The "hirak/prestissimo" plugin was skipped because it requires a Plugin API version ("^1.0.0") that does not match your Composer installation ("2.0.0"). You may need to run composer update with the "--no-plugins" option.
Running composer update hirak/prestissimo
Loading composer repositories with package information
Updating dependencies
Lock file operations: 0 installs, 0 updates, 1 removal
  - Removing hirak/prestissimo (0.3.10)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 0 installs, 0 updates, 1 removal
  - Removing hirak/prestissimo (0.3.10)
Generating autoload files
 Drush Version   :  8.4.8 

Killing REDACTED_appserver_1 ... 
Killing REDACTED_appserver_1 ... done
Pulling appserver_nginx (bitnami/nginx:1.16.1-debian-10-r106)...
1.16.1-debian-10-r106: Pulling from bitnami/nginx
f741ee3cf64f: Pull complete
7916b15bd5f3: Pull complete
9f7e47b9e838: Pull complete
a36750541329: Pull complete
d3a340251ce2: Pull complete
988248eb6cc1: Pull complete
9a2ff3c03e1e: Pull complete
54d6837b2194: Pull complete
628995fd0156: Pull complete
963c7f099cc8: Pull complete
551fec98430e: Pull complete
Digest: sha256:5ebe2a3e9d7cf377bfe6a9eca7cde7fb448bbacad692121b7fc8693fef95ac58
Status: Downloaded newer image for bitnami/nginx:1.16.1-debian-10-r106
Creating REDACTED_database_1        ... done
Creating REDACTED_appserver_nginx_1 ... done
Creating REDACTED_cache_1           ... done
Creating REDACTED_index_1           ... done
Starting REDACTED_appserver_1       ... done
Creating REDACTED_edge_1            ... done
Creating REDACTED_edge_ssl_1        ... done
Waiting until database service is ready...
Scanning to determine which services are ready... Please standby...
usermod: no changes
usermod: no changes

   ___                      __        __        __     __        ______
  / _ )___  ___  __ _  ___ / /  ___ _/ /_____ _/ /__ _/ /_____ _/ / / /
 / _  / _ \/ _ \/  ' \(_-</ _ \/ _ `/  '_/ _ `/ / _ `/  '_/ _ `/_/_/_/ 
/____/\___/\___/_/_/_/___/_//_/\_,_/_/\_\\_,_/_/\_,_/_/\_\\_,_(_|_|_)  
                                                                       

Your app has started up correctly.
Here are some vitals:

REDACTED
```

I removed the site infos to not expose the site name.  I did a '`lando pull` as well and it worked as expected.  So my guess something is sad on your system itself.

The best way to remedy that is to purge everything.  I can show you how (if you don't know), but you will lose all your local setups.  Are you in a position to do that?

**Comment by leorawe** [comment]
Yes, can you write up how to do a complete purge? It is on a computer that only has that site.

**Comment by leorawe** [comment]
It would not let me when I tried to double click on the uninstall script.

My error is because my old .lando.yml has some excludes at the bottom. It probably wasn't adding in all the proper files, and it was using the old .lando.yml because I had given the same name to the new one.

I rebuilt my old site using the old lando without any issues. I am going to follow Greg's instructions for a new method of conversion from 8 to 9. I appreciate your patience in helping me figuring out the issue.
